apps/trading/client-pages/swap/swap-form.tsx

```diff
@@ -0,0 +1,201 @@
+);
+
+export const PriceImpactInput = ({
+  value,
+  onValueChange,
+}: {
+  value: string;
+  onValueChange: (value: string) => void;
+}) => {
+  const t = useT();
+  const [, setPriceImpactType] = useState<'custom' | 'auto'>('custom');
+  const autoValues = ['0.1', '0.5', '1.0'];
+
+  return (
+    <div className="mb-4">
+      <div className="flex justify-between items-center mb-1 mt-2 text-gray-500 text-sm">
+        <span>{t('Price impact tolerance')}</span>
+      </div>
+      <div className="flex items-center">
+        <span className="w-16 h-10 rounded-lg mr-2 text-center text-md">
+          {value || ''} %
+        </span>
+      </div>
+      <div className="flex items-center flex-wrap">
+        {autoValues.map((val) => (
+          <button
+            key={val}
+            type="button"
+            value={value}
+            onClick={() => {
+              onValueChange(val);
+              setPriceImpactType('auto');
+            }}
+            className={classNames(
+              'h-8 text-md dark:bg-vega-cdark-500 bg-vega-clight-500 p-2 rounded-lg mr-2 text-center text-sm',
+              {
+                'dark:bg-vega-cdark-700 bg-vega-clight-700': val === value,
+              }
+            )}
+          >
+            {val} %
+          </button>
+        ))}
+
+        <div className="flex flex-1">
+          <p className="pt-2 pr-2 text-sm">{t('Custom')}: </p>
+          <TradingInput
+            type="number"
+            value={value}
+            onChange={(e) => {
+              onValueChange(e.target.value);
+              setPriceImpactType('custom');
+            }}
+            appendElement="%"
+            className="h-10 text-md dark:bg-vega-cdark-800 bg-vega-clight-500 p-2 rounded-lg mr-2 text-center"
+          />
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export const DropdownAsset = ({
+  assetId,
+  onSelect,
+  assets,
+}: {
+  assetId?: string;
+  onSelect: (asset: AssetFieldsFragment) => void;
+  assets?: Record<string, AssetFieldsFragment>;
+}) => {
+  const { chainId } = useChainId();
+  const asset = assetId ? assets?.[assetId] : null;
+  return (
+    <DropdownMenu
+      trigger={
+        <DropdownMenuTrigger
+          asChild
+          className="flex items-center px-2 py-2 border-gray-400 border rounded-full h-12"
+        >
+          {asset && <EmblemByAsset asset={asset.id} vegaChain={chainId} />}
+          <span className="pl-2">{asset ? asset.symbol : 'Select coin'}</span>
+          <VegaIcon
+            name={VegaIconNames.CHEVRON_DOWN}
+            size={14}
+            className="w-5 h-5 ml-2 flex items-center justify-center"
+          />
+        </DropdownMenuTrigger>
+      }
+    >
+      <DropdownMenuContent className="bg-gray-700 rounded-md mt-2">
+        {assets &&
+          Object.values(assets).map((asset) => (
+            <DropdownMenuItem
+              onClick={(e) => {
+                e.preventDefault();
+                e.stopPropagation();
```

Comment:
I'd much prefer if the dropdown closed on selection of the asset. Its quite annoying it stays open

Reply:
updated